azure-iot-device/azure/iot/device/common/mqtt_transport.py

```diff
@@ -264,14 +264,20 @@ def on_message(client, userdata, mqtt_message):
         mqtt_client.on_publish = on_publish
         mqtt_client.on_message = on_message
 
+        # Set paho automatic-reconnect delay to 2 hours.  Ideally we would turn
+        # paho auto-reconnect off entirely, but this is the best we can do.  Without
```

Comment:
>but this is the best we can do. [](start = 44, length = 31)

Seriously? Paho doesn't let you do this??

The more I learn about Paho...

Reply:
It sort of does.  You can turn off auto-reconnect by making your own IO thread.  In a perfect world with infinite time, that's what I would have done.

---
In reply to: [399533440](https://github.com/Azure/azure-iot-sdk-python/pull/516#discussion_r399533440) [](ancestors = 399533440)